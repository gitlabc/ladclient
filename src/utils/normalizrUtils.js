import { normalize, schema } from 'normalizr';

export function getRaceEvents(originalData) {
    const eventSchema = new schema.Entity('events', {}, {
        processStrategy: (value, parent, key) => {
            return { ...value, meeting_id: parent.id, type: parent.type };
        }
    });
    const meetingSchema = new schema.Entity('meetings', {
        events: [eventSchema],
    });
    const normalizedData = normalize(originalData, [meetingSchema]);
    return normalizedData;
}

