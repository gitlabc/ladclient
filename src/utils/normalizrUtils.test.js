import { normalize, schema } from 'normalizr';
import { getRaceEvents } from './normalizrUtils';

import * as meetingsPackage from './meetings.json';

describe('getRaceEvents', () => {

    test('should return a flat events', () => {
        let meetings = meetingsPackage.updates[0].data;
        let events = getRaceEvents(meetings);
        expect(events).toMatchSnapshot();
    });
});
