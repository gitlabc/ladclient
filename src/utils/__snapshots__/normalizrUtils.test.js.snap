// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRaceEvents should return a flat events 1`] = `
Object {
  "entities": Object {
    "events": Object {
      "11378": Object {
        "description": "Carter Knolls",
        "id": 11378,
        "meeting_id": 1976,
        "outcome": 1499258946,
        "race_num": 7,
        "status": "open",
      },
      "11560": Object {
        "description": "Adrienne Plaza",
        "id": 11560,
        "meeting_id": 3475,
        "outcome": 1499257394,
        "race_num": 3,
        "status": "open",
      },
      "13046": Object {
        "description": "Annamae Gardens",
        "id": 13046,
        "meeting_id": 5545,
        "outcome": 1499257204,
        "race_num": 2,
        "status": "open",
      },
      "13171": Object {
        "description": "Olson Junction",
        "id": 13171,
        "meeting_id": 5545,
        "outcome": 1499258927,
        "race_num": 7,
        "status": "open",
      },
      "13548": Object {
        "description": "Ortiz Shores",
        "id": 13548,
        "meeting_id": 8778,
        "outcome": 1499258855,
        "race_num": 6,
        "status": "open",
      },
      "14707": Object {
        "description": "Beahan Pass",
        "id": 14707,
        "meeting_id": 1166,
        "outcome": 1499259717,
        "race_num": 5,
        "status": "open",
      },
      "15167": Object {
        "description": "Reinger Common",
        "id": 15167,
        "meeting_id": 3751,
        "outcome": 1499260352,
        "race_num": 10,
        "status": "open",
      },
      "15246": Object {
        "description": "Effertz Spur",
        "id": 15246,
        "meeting_id": 2488,
        "outcome": 1499257944,
        "race_num": 6,
        "status": "open",
      },
      "15612": Object {
        "description": "Runolfsdottir Ports",
        "id": 15612,
        "meeting_id": 1976,
        "outcome": 1499257508,
        "race_num": 3,
        "status": "open",
      },
      "16689": Object {
        "description": "Kiehn Summit",
        "id": 16689,
        "meeting_id": 2488,
        "outcome": 1499259023,
        "race_num": 8,
        "status": "open",
      },
      "17976": Object {
        "description": "Crist Walks",
        "id": 17976,
        "meeting_id": 1976,
        "outcome": 1499259995,
        "race_num": 9,
        "status": "open",
      },
      "18440": Object {
        "description": "Turner Inlet",
        "id": 18440,
        "meeting_id": 8116,
        "outcome": 1499259573,
        "race_num": 8,
        "status": "open",
      },
      "19125": Object {
        "description": "Carlie Crossroad",
        "id": 19125,
        "meeting_id": 1166,
        "outcome": 1499258503,
        "race_num": 4,
        "status": "open",
      },
      "19601": Object {
        "description": "Moen Village",
        "id": 19601,
        "meeting_id": 2488,
        "outcome": 1499257128,
        "race_num": 3,
        "status": "open",
      },
      "20860": Object {
        "description": "Green Parkway",
        "id": 20860,
        "meeting_id": 8778,
        "outcome": 1499257834,
        "race_num": 4,
        "status": "open",
      },
      "21023": Object {
        "description": "Littel Orchard",
        "id": 21023,
        "meeting_id": 1976,
        "outcome": 1499257445,
        "race_num": 2,
        "status": "open",
      },
      "21693": Object {
        "description": "Kunze Vista",
        "id": 21693,
        "meeting_id": 8116,
        "outcome": 1499260209,
        "race_num": 10,
        "status": "open",
      },
      "22025": Object {
        "description": "Adolf Landing",
        "id": 22025,
        "meeting_id": 1976,
        "outcome": 1499259737,
        "race_num": 8,
        "status": "open",
      },
      "22716": Object {
        "description": "Lakin Vista",
        "id": 22716,
        "meeting_id": 5081,
        "outcome": 1499259414,
        "race_num": 8,
        "status": "open",
      },
      "24214": Object {
        "description": "Thompson Valleys",
        "id": 24214,
        "meeting_id": 2488,
        "outcome": 1499257179,
        "race_num": 4,
        "status": "open",
      },
      "24651": Object {
        "description": "Legros Extensions",
        "id": 24651,
        "meeting_id": 1166,
        "outcome": 1499260084,
        "race_num": 7,
        "status": "open",
      },
      "24784": Object {
        "description": "Torp Field",
        "id": 24784,
        "meeting_id": 5545,
        "outcome": 1499259208,
        "race_num": 8,
        "status": "open",
      },
      "25412": Object {
        "description": "Jessy Fields",
        "id": 25412,
        "meeting_id": 8116,
        "outcome": 1499258314,
        "race_num": 3,
        "status": "open",
      },
      "26334": Object {
        "description": "Rutherford Place",
        "id": 26334,
        "meeting_id": 8116,
        "outcome": 1499258108,
        "race_num": 2,
        "status": "open",
      },
      "26782": Object {
        "description": "Lakin Landing",
        "id": 26782,
        "meeting_id": 7407,
        "outcome": 1499258436,
        "race_num": 6,
        "status": "open",
      },
      "28338": Object {
        "description": "Beahan Grove",
        "id": 28338,
        "meeting_id": 1976,
        "outcome": 1499258791,
        "race_num": 6,
        "status": "open",
      },
      "28826": Object {
        "description": "Boyer Meadows",
        "id": 28826,
        "meeting_id": 8116,
        "outcome": 1499259481,
        "race_num": 7,
        "status": "open",
      },
      "29777": Object {
        "description": "Tromp Ports",
        "id": 29777,
        "meeting_id": 5081,
        "outcome": 1499257379,
        "race_num": 2,
        "status": "open",
      },
      "30458": Object {
        "description": "Helmer Forge",
        "id": 30458,
        "meeting_id": 8116,
        "outcome": 1499258934,
        "race_num": 6,
        "status": "open",
      },
      "31043": Object {
        "description": "Amira Well",
        "id": 31043,
        "meeting_id": 5081,
        "outcome": 1499257592,
        "race_num": 5,
        "status": "open",
      },
      "31636": Object {
        "description": "Emard Knolls",
        "id": 31636,
        "meeting_id": 3751,
        "outcome": 1499258888,
        "race_num": 6,
        "status": "open",
      },
      "31960": Object {
        "description": "Senger Road",
        "id": 31960,
        "meeting_id": 5545,
        "outcome": 1499258000,
        "race_num": 5,
        "status": "open",
      },
      "32323": Object {
        "description": "Bahringer Locks",
        "id": 32323,
        "meeting_id": 1976,
        "outcome": 1499257811,
        "race_num": 4,
        "status": "open",
      },
      "32352": Object {
        "description": "Greg Locks",
        "id": 32352,
        "meeting_id": 7407,
        "outcome": 1499258156,
        "race_num": 5,
        "status": "open",
      },
      "32679": Object {
        "description": "Burnice Expressway",
        "id": 32679,
        "meeting_id": 3475,
        "outcome": 1499259880,
        "race_num": 9,
        "status": "open",
      },
      "32831": Object {
        "description": "Frances Dam",
        "id": 32831,
        "meeting_id": 3475,
        "outcome": 1499257667,
        "race_num": 4,
        "status": "open",
      },
      "33093": Object {
        "description": "Bradtke Pass",
        "id": 33093,
        "meeting_id": 3751,
        "outcome": 1499259324,
        "race_num": 7,
        "status": "open",
      },
      "33766": Object {
        "description": "Runolfsdottir Light",
        "id": 33766,
        "meeting_id": 5545,
        "outcome": 1499260125,
        "race_num": 10,
        "status": "open",
      },
      "33827": Object {
        "description": "Madonna Lakes",
        "id": 33827,
        "meeting_id": 8116,
        "outcome": 1499257908,
        "race_num": 1,
        "status": "open",
      },
      "34641": Object {
        "description": "Verda Dam",
        "id": 34641,
        "meeting_id": 8116,
        "outcome": 1499258387,
        "race_num": 4,
        "status": "open",
      },
      "35261": Object {
        "description": "Bruen Fields",
        "id": 35261,
        "meeting_id": 5081,
        "outcome": 1499258604,
        "race_num": 6,
        "status": "open",
      },
      "37298": Object {
        "description": "Rogahn Mission",
        "id": 37298,
        "meeting_id": 7407,
        "outcome": 1499257381,
        "race_num": 2,
        "status": "open",
      },
      "37865": Object {
        "description": "Hoeger Meadows",
        "id": 37865,
        "meeting_id": 7407,
        "outcome": 1499259878,
        "race_num": 8,
        "status": "open",
      },
      "39308": Object {
        "description": "O'Kon Hill",
        "id": 39308,
        "meeting_id": 3475,
        "outcome": 1499257181,
        "race_num": 1,
        "status": "open",
      },
      "39422": Object {
        "description": "Lexus Cliff",
        "id": 39422,
        "meeting_id": 2488,
        "outcome": 1499259830,
        "race_num": 9,
        "status": "open",
      },
      "40506": Object {
        "description": "Cathrine Drives",
        "id": 40506,
        "meeting_id": 8778,
        "outcome": 1499259922,
        "race_num": 8,
        "status": "open",
      },
      "40585": Object {
        "description": "Autumn Knolls",
        "id": 40585,
        "meeting_id": 8778,
        "outcome": 1499257150,
        "race_num": 1,
        "status": "open",
      },
      "40809": Object {
        "description": "Trey Via",
        "id": 40809,
        "meeting_id": 3475,
        "outcome": 1499257762,
        "race_num": 5,
        "status": "open",
      },
      "41420": Object {
        "description": "Annalise Fords",
        "id": 41420,
        "meeting_id": 3751,
        "outcome": 1499258213,
        "race_num": 4,
        "status": "open",
      },
      "42183": Object {
        "description": "Amos Gardens",
        "id": 42183,
        "meeting_id": 5545,
        "outcome": 1499257121,
        "race_num": 1,
        "status": "open",
      },
      "43730": Object {
        "description": "Jakubowski Turnpike",
        "id": 43730,
        "meeting_id": 3751,
        "outcome": 1499257002,
        "race_num": 1,
        "status": "open",
      },
      "45023": Object {
        "description": "Sandrine Court",
        "id": 45023,
        "meeting_id": 8778,
        "outcome": 1499259825,
        "race_num": 7,
        "status": "open",
      },
      "46707": Object {
        "description": "Jason Road",
        "id": 46707,
        "meeting_id": 3475,
        "outcome": 1499258961,
        "race_num": 6,
        "status": "open",
      },
      "48065": Object {
        "description": "Mayra Flats",
        "id": 48065,
        "meeting_id": 7407,
        "outcome": 1499260165,
        "race_num": 9,
        "status": "open",
      },
      "48611": Object {
        "description": "Earlene Mountains",
        "id": 48611,
        "meeting_id": 5081,
        "outcome": 1499257526,
        "race_num": 4,
        "status": "open",
      },
      "48625": Object {
        "description": "Evert Oval",
        "id": 48625,
        "meeting_id": 5081,
        "outcome": 1499257508,
        "race_num": 3,
        "status": "open",
      },
      "48957": Object {
        "description": "Mitchell Ports",
        "id": 48957,
        "meeting_id": 8116,
        "outcome": 1499258693,
        "race_num": 5,
        "status": "open",
      },
      "49864": Object {
        "description": "Kulas Isle",
        "id": 49864,
        "meeting_id": 3751,
        "outcome": 1499259743,
        "race_num": 9,
        "status": "open",
      },
      "50506": Object {
        "description": "Treutel Path",
        "id": 50506,
        "meeting_id": 5545,
        "outcome": 1499257557,
        "race_num": 4,
        "status": "open",
      },
      "50707": Object {
        "description": "Bailey Prairie",
        "id": 50707,
        "meeting_id": 3751,
        "outcome": 1499257138,
        "race_num": 2,
        "status": "open",
      },
      "51125": Object {
        "description": "Wilburn Heights",
        "id": 51125,
        "meeting_id": 7407,
        "outcome": 1499259393,
        "race_num": 7,
        "status": "open",
      },
      "51400": Object {
        "description": "Lia Plains",
        "id": 51400,
        "meeting_id": 1166,
        "outcome": 1499260406,
        "race_num": 8,
        "status": "open",
      },
      "51957": Object {
        "description": "Jalen Courts",
        "id": 51957,
        "meeting_id": 5081,
        "outcome": 1499258929,
        "race_num": 7,
        "status": "open",
      },
      "51992": Object {
        "description": "Zieme Ranch",
        "id": 51992,
        "meeting_id": 8778,
        "outcome": 1499260208,
        "race_num": 9,
        "status": "open",
      },
      "52577": Object {
        "description": "Greenfelder Land",
        "id": 52577,
        "meeting_id": 2488,
        "outcome": 1499257498,
        "race_num": 5,
        "status": "open",
      },
      "53106": Object {
        "description": "Aubrey Landing",
        "id": 53106,
        "meeting_id": 1976,
        "outcome": 1499260208,
        "race_num": 10,
        "status": "open",
      },
      "55138": Object {
        "description": "Little Drives",
        "id": 55138,
        "meeting_id": 1166,
        "outcome": 1499257223,
        "race_num": 1,
        "status": "open",
      },
      "58218": Object {
        "description": "Dorothea Fork",
        "id": 58218,
        "meeting_id": 8778,
        "outcome": 1499257470,
        "race_num": 2,
        "status": "open",
      },
      "59441": Object {
        "description": "Harvey Square",
        "id": 59441,
        "meeting_id": 7407,
        "outcome": 1499257023,
        "race_num": 1,
        "status": "open",
      },
      "60041": Object {
        "description": "Lang Fork",
        "id": 60041,
        "meeting_id": 5545,
        "outcome": 1499259819,
        "race_num": 9,
        "status": "open",
      },
      "61405": Object {
        "description": "Jayne Points",
        "id": 61405,
        "meeting_id": 7407,
        "outcome": 1499257831,
        "race_num": 3,
        "status": "open",
      },
      "62249": Object {
        "description": "Audra Turnpike",
        "id": 62249,
        "meeting_id": 1166,
        "outcome": 1499260531,
        "race_num": 10,
        "status": "open",
      },
      "62445": Object {
        "description": "Gianni Turnpike",
        "id": 62445,
        "meeting_id": 1976,
        "outcome": 1499258691,
        "race_num": 5,
        "status": "open",
      },
      "62566": Object {
        "description": "Katelin Locks",
        "id": 62566,
        "meeting_id": 2488,
        "outcome": 1499257002,
        "race_num": 1,
        "status": "open",
      },
      "62842": Object {
        "description": "Goldner Junction",
        "id": 62842,
        "meeting_id": 7407,
        "outcome": 1499260363,
        "race_num": 10,
        "status": "open",
      },
      "63044": Object {
        "description": "Eloy Pike",
        "id": 63044,
        "meeting_id": 1976,
        "outcome": 1499257023,
        "race_num": 1,
        "status": "open",
      },
      "63607": Object {
        "description": "Lueilwitz Lane",
        "id": 63607,
        "meeting_id": 8778,
        "outcome": 1499260305,
        "race_num": 10,
        "status": "open",
      },
      "65255": Object {
        "description": "Sporer Views",
        "id": 65255,
        "meeting_id": 5081,
        "outcome": 1499260363,
        "race_num": 10,
        "status": "open",
      },
      "67371": Object {
        "description": "Lang Centers",
        "id": 67371,
        "meeting_id": 1166,
        "outcome": 1499257681,
        "race_num": 2,
        "status": "open",
      },
      "67390": Object {
        "description": "Hills Park",
        "id": 67390,
        "meeting_id": 3475,
        "outcome": 1499259166,
        "race_num": 7,
        "status": "open",
      },
      "68724": Object {
        "description": "Koelpin Port",
        "id": 68724,
        "meeting_id": 1166,
        "outcome": 1499257940,
        "race_num": 3,
        "status": "open",
      },
      "69285": Object {
        "description": "Johnston Meadow",
        "id": 69285,
        "meeting_id": 5545,
        "outcome": 1499257377,
        "race_num": 3,
        "status": "open",
      },
      "72363": Object {
        "description": "Bridgette Summit",
        "id": 72363,
        "meeting_id": 2488,
        "outcome": 1499260417,
        "race_num": 10,
        "status": "open",
      },
      "72481": Object {
        "description": "Glover Cliffs",
        "id": 72481,
        "meeting_id": 8778,
        "outcome": 1499258329,
        "race_num": 5,
        "status": "open",
      },
      "73080": Object {
        "description": "Andrew Inlet",
        "id": 73080,
        "meeting_id": 3475,
        "outcome": 1499259700,
        "race_num": 8,
        "status": "open",
      },
      "74851": Object {
        "description": "Zora Run",
        "id": 74851,
        "meeting_id": 8116,
        "outcome": 1499259960,
        "race_num": 9,
        "status": "open",
      },
      "74900": Object {
        "description": "Collier Squares",
        "id": 74900,
        "meeting_id": 8778,
        "outcome": 1499257742,
        "race_num": 3,
        "status": "open",
      },
      "76423": Object {
        "description": "Mertz Loaf",
        "id": 76423,
        "meeting_id": 2488,
        "outcome": 1499257005,
        "race_num": 2,
        "status": "open",
      },
      "76518": Object {
        "description": "Karolann Orchard",
        "id": 76518,
        "meeting_id": 7407,
        "outcome": 1499257925,
        "race_num": 4,
        "status": "open",
      },
      "80121": Object {
        "description": "Sipes Springs",
        "id": 80121,
        "meeting_id": 3475,
        "outcome": 1499257263,
        "race_num": 2,
        "status": "open",
      },
      "81025": Object {
        "description": "Terrence Crescent",
        "id": 81025,
        "meeting_id": 3475,
        "outcome": 1499259947,
        "race_num": 10,
        "status": "open",
      },
      "81575": Object {
        "description": "Runolfsson Terrace",
        "id": 81575,
        "meeting_id": 3751,
        "outcome": 1499258733,
        "race_num": 5,
        "status": "open",
      },
      "82100": Object {
        "description": "Newell Mountains",
        "id": 82100,
        "meeting_id": 5081,
        "outcome": 1499259466,
        "race_num": 9,
        "status": "open",
      },
      "83184": Object {
        "description": "Lila Vista",
        "id": 83184,
        "meeting_id": 1166,
        "outcome": 1499259847,
        "race_num": 6,
        "status": "open",
      },
      "84655": Object {
        "description": "Schiller Mills",
        "id": 84655,
        "meeting_id": 1166,
        "outcome": 1499260512,
        "race_num": 9,
        "status": "open",
      },
      "84867": Object {
        "description": "Stehr Parkway",
        "id": 84867,
        "meeting_id": 3751,
        "outcome": 1499259738,
        "race_num": 8,
        "status": "open",
      },
      "85204": Object {
        "description": "Steve Squares",
        "id": 85204,
        "meeting_id": 2488,
        "outcome": 1499258479,
        "race_num": 7,
        "status": "open",
      },
      "87184": Object {
        "description": "Okuneva Ramp",
        "id": 87184,
        "meeting_id": 5545,
        "outcome": 1499258205,
        "race_num": 6,
        "status": "open",
      },
      "88526": Object {
        "description": "Lexus Village",
        "id": 88526,
        "meeting_id": 3751,
        "outcome": 1499258149,
        "race_num": 3,
        "status": "open",
      },
      "89783": Object {
        "description": "Eichmann Wall",
        "id": 89783,
        "meeting_id": 5081,
        "outcome": 1499257325,
        "race_num": 1,
        "status": "open",
      },
    },
    "meetings": Object {
      "1166": Object {
        "country": "AU",
        "date": "2017-07-05",
        "events": Array [
          55138,
          67371,
          68724,
          19125,
          14707,
          83184,
          24651,
          51400,
          84655,
          62249,
        ],
        "id": 1166,
        "name": "Colemanfort",
        "type": "T",
      },
      "1976": Object {
        "country": "AU",
        "date": "2017-07-06",
        "events": Array [
          63044,
          21023,
          15612,
          32323,
          62445,
          28338,
          11378,
          22025,
          17976,
          53106,
        ],
        "id": 1976,
        "name": "West Otisborough",
        "type": "G",
      },
      "2488": Object {
        "country": "AU",
        "date": "2017-07-06",
        "events": Array [
          62566,
          76423,
          19601,
          24214,
          52577,
          15246,
          85204,
          16689,
          39422,
          72363,
        ],
        "id": 2488,
        "name": "West Willy",
        "type": "G",
      },
      "3475": Object {
        "country": "AU",
        "date": "2017-07-05",
        "events": Array [
          39308,
          80121,
          11560,
          32831,
          40809,
          46707,
          67390,
          73080,
          32679,
          81025,
        ],
        "id": 3475,
        "name": "Bonniebury",
        "type": "H",
      },
      "3751": Object {
        "country": "AU",
        "date": "2017-07-06",
        "events": Array [
          43730,
          50707,
          88526,
          41420,
          81575,
          31636,
          33093,
          84867,
          49864,
          15167,
        ],
        "id": 3751,
        "name": "North Vitofort",
        "type": "T",
      },
      "5081": Object {
        "country": "AU",
        "date": "2017-07-06",
        "events": Array [
          89783,
          29777,
          48625,
          48611,
          31043,
          35261,
          51957,
          22716,
          82100,
          65255,
        ],
        "id": 5081,
        "name": "Stephanietown",
        "type": "G",
      },
      "5545": Object {
        "country": "AU",
        "date": "2017-07-05",
        "events": Array [
          42183,
          13046,
          69285,
          50506,
          31960,
          87184,
          13171,
          24784,
          60041,
          33766,
        ],
        "id": 5545,
        "name": "Jamaalmouth",
        "type": "H",
      },
      "7407": Object {
        "country": "AU",
        "date": "2017-07-06",
        "events": Array [
          59441,
          37298,
          61405,
          76518,
          32352,
          26782,
          51125,
          37865,
          48065,
          62842,
        ],
        "id": 7407,
        "name": "North Garry",
        "type": "H",
      },
      "8116": Object {
        "country": "AU",
        "date": "2017-07-05",
        "events": Array [
          33827,
          26334,
          25412,
          34641,
          48957,
          30458,
          28826,
          18440,
          74851,
          21693,
        ],
        "id": 8116,
        "name": "Port Leslieton",
        "type": "H",
      },
      "8778": Object {
        "country": "AU",
        "date": "2017-07-06",
        "events": Array [
          40585,
          58218,
          74900,
          20860,
          72481,
          13548,
          45023,
          40506,
          51992,
          63607,
        ],
        "id": 8778,
        "name": "Leannonborough",
        "type": "T",
      },
    },
  },
  "result": Array [
    3751,
    1166,
    8778,
    1976,
    2488,
    5081,
    8116,
    7407,
    3475,
    5545,
  ],
}
`;
